appinventor/components/src/com/google/appinventor/components/runtime/ChatBot.java

```diff
@@ -634,6 +673,42 @@ private synchronized void ensureSslSockFactory() {
     }
   }
 
+  /*
+   * Parse response from the Chatbot. Extract the image data from the
+   * returned protocol buffer, if provided. Call either GotResponse,
+   * or GotResponseWithImage depending on whether or not an image was
+   * returned.
+   *
+   * @param parsed protocol buffer returned from proxy
+   * @returns Void
+   */
+  private void parseResponse(ChatBotToken.response response) {
+    String responseText = response.getAnswer();
+    byte[] responseImage = response.getOutputimage().toByteArray();
+    if (DEBUG) {
+      Log.d(LOG_TAG, "parseResponse: responseImage = " + responseImage);
+    }
+    if (responseImage != null && responseImage.length > 0) {
+      try {
+        File outFile = getOutputFile();
+        FileOutputStream out = new FileOutputStream(outFile);
+        try {
+          out.write(responseImage);
+          out.flush();
+        } finally {
+          out.close();
+        }
+        String responseImageUri = Uri.fromFile(outFile).toString();
```

Comment:
Since we already have `java.net.URL` imported, are we able to use that instead of having to rely on `android.net.Uri`?

Reply:
I'm more comfortable leaving this alone. I believe android.net.Uri is optimized for Android.